extension-ui/background.js

```diff
@@ -0,0 +1,32 @@
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.storage.local.set({ blockedSites: [] });
+});
+
+chrome.webNavigation.onBeforeNavigate.addListener((details) => {
+  const url = new URL(details.url);
+  if (url.protocol === 'chrome:' || url.protocol === 'about:') {
+    return;
+  }
+ 
+  chrome.storage.local.get(["blockedSites"], (data) => {
```

Comment:
wherever you used chrome.storage.local.get, please save in to variable and use it. only if it's empty, get the data again from the local storage